crates/polars-arrow/src/legacy/kernels/take_agg/mod.rs

```diff
@@ -12,7 +12,7 @@ use crate::types::NativeType;
 
 /// Take kernel for single chunk without nulls and an iterator as index.
 /// # Safety
-/// caller must ensure iterators indexes are in bounds
+/// caller must ensure iterators are nonempty and indexes are in bounds
```

Comment:
```suggestion
/// caller must ensure iterators are non-empty and indexes are in bounds
```

Reply:
https://gcc.gnu.org/legacy-ml/gcc/2001-10/msg00610.html

Reply:
Nonzero and non-empty are not the same thing ;)
https://dictionary.cambridge.org/dictionary/english/non-empty?q=nonempty

Down to personal preference at this point I suppose; seems both are considered valid at this point, though the Cambridge dictionary has the primary definition as "non-empty" with "nonempty" being an accepted secondary variation.